src/coreclr/jit/hwintrinsic.h

```diff
@@ -587,6 +590,27 @@ struct HWIntrinsicInfo
         HWIntrinsicFlag flags = lookupFlags(id);
         return (flags & HW_Flag_EmbBroadcastCompatible) != 0;
     }
+
+    static bool IsEmbRoundingCompatible(NamedIntrinsic id)
+    {
+        HWIntrinsicFlag flags = lookupFlags(id);
+        return (flags & HW_Flag_EmbRoundingCompatible) != 0;
+    }
+
+    static size_t EmbRoundingArgPos(NamedIntrinsic id)
+    {
+        // This helper function returns the expected position,
+        // where the embedded rounding control argument should be.
+        assert(IsEmbRoundingCompatible(id));
+        switch (id)
+        {
+            case NI_AVX512F_Add:
+                return 3;
```

Comment:
If it isn't, we should be able to effectively use `node->GetOperandCount() - 1` instead.

Reply:
This acts mostly like a table which records the expected number of operands when embedded rounding is activated, as we need an indication to tell if this feature is enabled or it can be just lowered as usual. 

The reason I add this is that as now `Avx512F.Add()` can take 2 or 3 inputs, 1 extra input as `FloatRoundingMode`, on the table in `hwintrinsiclistxarch.h`, we put `-1` at `NumArg` to accommodate multiply choices on input numbers. So we don't have how many args are expected in most of the cases, and as we will be handling some type casting and scalar intrinsics, I suppose only checking if the type of the last argument is constant int might not enough.

 In our case, `node->GetOperandCount()` would dynamically return 2 or 3 based on the number of inputs, and we might need some information to tell the difference.